systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -253,4 +258,182 @@ void testKafkaNodePoolFeatureGate(ExtensionContext extensionContext) {
         RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), kafkaSelector, 3, kafkaPods);
 
     }
+
+    /**
+     * @description This test case verifies transfer of Kafka Cluster from and to management by KafkaNodePool, by creating corresponding Kafka and KafkaNodePool custom resources
+     * and manipulating according kafka annotation.
+     *
+     * @steps
+     * 1. - Deploy Kafka with annotated to enable management by KafkaNodePool, and KafkaNodePool targeting given Kafka Cluster.
+     *    - Kafka is deployed, KafkaNodePool custom resource is targeting Kafka Cluster as expected.
+     * 2. - Modify KafkaNodePool by increasing number of Kafka Replicas.
+     *    - Number of Kafka Pods is increased to match specification from KafkaNodePool
+     * 3. - Produce and consume messages in given Kafka Cluster.
+     *    - Clients can produce and consume messages.
+     * 4. - Modify Kafka custom resource annotation strimzi.io/node-pool to disable management by KafkaNodePool.
+     *    - StrimziPodSet is modified, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) are ignored.
+     * 5. - Produce and consume messages in given Kafka Cluster.
+     *    - Clients can produce and consume messages.
+     * 6. - Modify Kafka custom resource annotation strimzi.io/node-pool to enable management by KafkaNodePool.
+     *    - new StrimziPodSet is created, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) has priority over Kafka specification.
+     * 7. - Produce and consume messages in given Kafka Cluster.
+     *    - Clients can produce and consume messages.
+     *
+     * @usecase
+     * - kafka-node-pool
+     */
+    @IsolatedTest
+    @SuppressWarnings({"checkstyle:MethodLength"})
+    void testKafkaManagementTransferToAndFromKafkaNodePool(ExtensionContext extensionContext) {
+        assumeFalse(Environment.isKRaftModeEnabled());
+        assumeFalse(Environment.isOlmInstall() || Environment.isHelmInstall());
+
+        final TestStorage testStorage = new TestStorage(extensionContext);
+
+        List<EnvVar> coEnvVars = new ArrayList<>();
+        coEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+KafkaNodePools", null));
+
+        clusterOperator = this.clusterOperator.defaultInstallation(extensionContext)
+            .withExtraEnvVars(coEnvVars)
+            .withNamespace(Constants.CO_NAMESPACE)
+            .withBindingsNamespaces(Arrays.asList(Constants.CO_NAMESPACE, Environment.TEST_SUITE_NAMESPACE))
+            .createInstallation()
+            .runInstallation();
+
+        final int originalKafkaReplicaCount = 3;
+        final int nodePoolIncreasedKafkaReplicaCount = 5;
+        final String kafkaNodePoolName = "kafka";
+
+        // setup clients
+        KafkaClients clients = new KafkaClientsBuilder()
```

Comment:
```suggestion
        final KafkaClients clients = new KafkaClientsBuilder()
```

Reply:
makes sense, thanks! (same for all other finals)